src/functions-and-arrays.js

```diff
@@ -1,41 +1,162 @@
 // Iteration #1: Find the maximum
-function maxOfTwoNumbers() {}
+function maxOfTwoNumbers( a, b) {
+  if (a>b){
+    return a;
+  }
+  else if (a<b){
+    return b;
+  }
+  else{
+    return a;
+  }
+}
 
 
 
 // Iteration #2: Find longest word
 const words = ['mystery', 'brother', 'aviator', 'crocodile', 'pearl', 'orchard', 'crackpot'];
 
-function findLongestWord() {}
+function findLongestWord(wordArray) {
+  if (wordArray.length === 0){
+    return null;
+  }
+  if (wordArray.length === 1){
+    return wordArray[0];
+  }
+
+  let wordLength=0;
+  let wordOcurrence;
+  for(var i=0; i<= wordArray.length-1; i++){
+    if ( wordArray[i].length > wordLength ){
+      wordLength=wordArray[i].length;
+      wordOcurrence=wordArray[i];
+    }
+  }
+  return wordOcurrence;
+}
 
 
 
 // Iteration #3: Calculate the sum
 const numbers = [6, 12, 1, 18, 13, 16, 2, 1, 8, 10];
 
-function sumNumbers() {}
+function sumNumbers(arr) {
+  if (arr.length === 0){
+    return 0;
+  }
+  if (arr.length === 1){
+    return arr[0];
+  }
+  let sum=0;
+  for(var i=0; i<= arr.length-1; i++){
+    sum+=arr[i];
+  }
+  return sum;
+}
 
 
 
 // Iteration #3.1 Bonus:
-function sum() {}
+function sum(arr) {
+  if (arr.length === 0){
+    return 0;
+  }
+  if (arr.length === 1){
+    return arr[0];
+  }
+  let sum=0;
+  for(var i=0; i<= arr.length-1; i++){
+    let arrObjType=typeof(arr[i]);
+    if (arrObjType==="number"){
+      sum+=arr[i];
+    }
+    else if (arrObjType==="string"){
+      sum+=arr[i].length;
+    }
+    else if(arrObjType==="boolean"){
+      if (arr[i]===true){
+        sum+=1;
+      }
+      else{
+        sum+=0;
+      }
+    }
+    else{
+      throw new Error("Unsupported data type sir or ma'am");
+    }    
+  }
+  return sum;
+}
 
 
 
 // Iteration #4: Calculate the average
 // Level 1: Array of numbers
 const numbersAvg = [2, 6, 9, 10, 7, 4, 1, 9];
 
-function averageNumbers() {}
+function averageNumbers(arr) {
+  if (arr.length === 0){
+    return null;
+  }
+  if (arr.length === 1){
+    return arr[0];
+  }
+  let sum=0;
+  for(var i=0; i<= arr.length-1; i++){
+    sum+=arr[i];
+  }
+  return sum/arr.length;
+}
 
 
 // Level 2: Array of strings
 const wordsArr = ['seat', 'correspond', 'linen', 'motif', 'hole', 'smell', 'smart', 'chaos', 'fuel', 'palace'];
 
-function averageWordLength() { }
+function averageWordLength(arr) {
+  if (arr.length === 0){
+    return null;
+  }
+  if (arr.length === 1){
+    return arr[0].length;
+  }
+  let sum=0;
+  for(var i=0; i<= arr.length-1; i++){
+    sum+=arr[i].length;
+  }
+  return sum/arr.length;
+}
 
 // Bonus - Iteration #4.1
-function avg() {}
+function avg(arr) {
```

Comment:
and again ^^ you could re-use the summation logic you already wrote by calling that function.